unit-tests/unit-test-config.py

```diff
@@ -139,16 +139,37 @@ def find_includes( filepath ):
     return filelist
 
 def process_cpp( dir, builddir ):
+    global regex, required_tags, list_only, available_tags, available_tests
     found = []
     shareds = []
     statics = []
-    for f in find( dir, '(^|/)test-.*\.cpp$' ):
+    if regex:
+        pattern = re.compile( regex )
+    for f in file.find( dir, '(^|/)test-.*\.cpp$' ):
         testdir = os.path.splitext( f )[0]                          # "log/internal/test-all"  <-  "log/internal/test-all.cpp"
         testparent = os.path.dirname(testdir)                       # "log/internal"
         # Each CMakeLists.txt sits in its own directory
-        os.makedirs( builddir + '/' + testdir, exist_ok = True );   # "build/log/internal/test-all"
+        os.makedirs( builddir + '/' + testdir, exist_ok = True )    # "build/log/internal/test-all"
         # We need the project name unique: keep the path but make it nicer:
         testname = 'test-' + testparent.replace( '/', '-' ) + '-' + os.path.basename(testdir)[5:]   # "test-log-internal-all"
+
+        #TODO: 1. check if testname matches the regex if -r was used
+        #      2. if -t was used, create ExeTest using the testname and check if the tags fit
+        #    if either of these don't match, continue
+        if regex and not pattern.search( testname ):
+            continue
+
+        if required_tags or list_tags:
+            test = unittest.ExeTest( testname, None )
```

Comment:
We can also directly use `TestConfigAsTest`, right? But then we'd have to provide the regex? Hmm...
Please document the `None`

Reply:
It's weid, creating an `ExeTest` without an exe!
I think maybe we can abstract a `CppConfig` from `TestConfigAsText` and then use it directly here

Reply:
`TestConfigFromCpp`